azkaban-web-server/src/main/java/cloudflow/servlets/ExecutionServlet.java

```diff
@@ -143,32 +144,42 @@ protected void handleGet(HttpServletRequest req, HttpServletResponse resp, Sessi
   protected void handlePost(HttpServletRequest req, HttpServletResponse resp, Session session)
       throws IOException {
 
-    if (GET_ALL_EXECUTIONS_URI.match(req.getRequestURI(), new HashMap<>())) {
-      ExecutionParameters executionParameters = extractExecutionParameters(req, resp, session);
-      if (executionParameters == null) {
-        return;
-      }
-
-      String executionId;
-      try {
-        executionId = executionService.createExecution(executionParameters);
-      } catch (CloudFlowValidationException e) {
-        sendErrorResponse(resp, HttpServletResponse.SC_BAD_REQUEST, e.getMessage());
-        return;
-      } catch (CloudFlowException e) {
-        sendErrorResponse(resp, HttpServletResponse.SC_INTERNAL_SERVER_ERROR,
-            DEFAULT_500_ERROR_MESSAGE);
-        return;
-      }
-      logger.info(String.format("New execution of flow %s was successfully queued with id %s",
-          executionParameters.getFlowId(), executionId));
-      resp.setHeader("Location", GET_SINGLE_EXECUTION_URI.createURI(executionId));
-      sendResponse(resp, HttpServletResponse.SC_CREATED, new HashMap<>());
+    String requestURI = req.getRequestURI();
+    if (GET_ALL_EXECUTIONS_URI.match(requestURI, new HashMap<>())) {
+      handleCreateExecution(req, resp, session);
     } else {
       sendErrorResponse(resp, HttpServletResponse.SC_NOT_FOUND, DEFAULT_404_ERROR_MESSAGE);
     }
   }
 
+  private void handleCreateExecution(HttpServletRequest req, HttpServletResponse resp,
+      Session session) throws IOException {
+    ExecutionParameters executionParameters = extractExecutionParameters(req, resp, session);
+    if (executionParameters == null) {
+      return;
+    }
```

Comment:
Shouldn't this be an exception ? (or at least a `requireNonNull` kind of check)

Reply:
It's intentionally actually, to stop execution of subsequent code. `extracExecutionParameters` takes care of throwing the appropriate exceptions and constructing the correct responses but calling `sendResponse` is not final, execution of `handleCreateExecution` continues after this method returns.

Reply:
Could please add a comment reflecting this in the code ?

Reply:
ok